e2e-tests/serviceless-external-nodes/run

```diff
@@ -1,115 +1,65 @@
-  kubectl_bin run network --image wbitt/network-multitool
-}
-
-function start_external_cluster() {
-  desc "Start External Cluster in unmanaged mode"
-  create_namespace "$external_ns"
-  kubectl_bin config set-context "$(kubectl_bin config current-context)" --namespace="$external_ns"
-  kubectl_bin apply -f "$test_dir/conf/secrets.yaml"
-
-  apply_cluster "$test_dir/conf/external.yaml"
-
-  sleep 2
-  wait_for_running "$db_name-rs0" 1
-}
-
-function start_main_cluster() {
-  desc "Start Main Cluster"
-  create_namespace "$main_ns"
-  kubectl_bin config set-context "$(kubectl_bin config current-context)" --namespace="$main_ns"
-  kubectl_bin apply -f "$test_dir/conf/secrets.yaml"
-  apply_cluster "$test_dir/conf/main.yaml"
-
-  sleep 2
-  wait_for_running "$db_name-rs0" 1
-}
-
-function cleanup() {
-  destroy "$external_ns"
-  destroy "$main_ns"
-  destroy "$operator_ns"
-}
-
-function get_connect_url() {
-  echo "$db_name-rs0-0.$db_name-rs0.$main_ns.svc.cluster.local,$db_name-rs0-0.$db_name-rs0.$external_ns.svc.cluster.local,$db_name-rs0-1.$db_name-rs0.$external_ns.svc.cluster.local"
-}
-
-function exec_mongo() {
-  local command="$1"
-  local uri="$2"
-  local client_container
-  client_container=$(kubectl_bin get pods -n $client_ns --selector=name=psmdb-client -o 'jsonpath={.items[].metadata.name}')
-
-  kubectl_bin exec "${client_container}" -n "$client_ns" -- \
-    bash -c "printf '$command\n' | mongo --quiet $uri"
-
-}
-
-function print_rs_status() {
-  url=$(get_connect_url)
-  exec_mongo \
-    'rs.status().members.forEach(function(z){printjson(z.name);printjson(z.stateStr); })' \
-    "mongodb://clusterAdmin:clusterAdmin123456@$url"
-}
-
-function test_rs_formed() {
-  desc "Testing that replica set with external members is formed..."
-  # shellcheck disable=SC2154 # ignoring - tmp_dir is defined in functions - spaghetti anyone?
-  print_rs_status >"$tmp_dir/rs.txt"
-
-  diff "${test_dir}/compare/rs.txt" "$tmp_dir/rs.txt"
-
-}
-
-function main() {
-  deploy_operator_cw
-  deploy_client
-  start_main_cluster
-  start_external_cluster
-  sleep 10
-
-  test_rs_formed
-}
-
-main
+. "${test_dir}/../functions"
+
+if [[ ${IMAGE_MONGOD} == *"percona-server-mongodb-operator"* ]]; then
+	MONGO_VER=$(echo -n "${IMAGE_MONGOD}" | $sed -r 's/.*([0-9].[0-9])$/\1/')
+else
+	MONGO_VER=$(echo -n "${IMAGE_MONGOD}" | $sed -r 's/.*:([0-9]+\.[0-9]+).*$/\1/')
+fi
+unset OPERATOR_NS
+
+desc "Create main cluster"
+create_infra "$namespace"
+kubectl_bin apply \
+	-f "$conf_dir/client.yml" \
+	-f "$test_dir/conf/secrets.yml"
+
+apply_cluster "$test_dir/conf/main.yml"
+wait_for_running "$cluster-rs0" 1
+
+desc "Start External Cluster in unmanaged mode"
+kubectl_bin config set-context $(kubectl_bin config current-context) --namespace="$replica_namespace"
+create_namespace $replica_namespace 0
+deploy_operator
+
+kubectl_bin apply \
+	-f "$conf_dir/client.yml" \
+	-f "$test_dir/conf/secrets.yml"
+apply_cluster "$test_dir/conf/external.yml"
+
+sleep 30
```

Comment:
Is there something we can wait for instead of putting hard code sleep? And if we are putting sleep we need at least comment what we wait.

Reply:
fixed